repos/system_upgrade/el7toel8/actors/persistentnetnamesconfig/libraries/persistentnetnamesconfig.py

```diff
@@ -23,3 +33,57 @@ def generate_link_file(interface):
         f.write(LINK_FILE_TEMPLATE.format(interface.mac, interface.name))
 
     return link_file
+
+
+def process():
+    rhel7_ifaces = next(api.consume(PersistentNetNamesFacts)).interfaces
+    rhel8_ifaces = next(api.consume(PersistentNetNamesFactsInitramfs)).interfaces
+
+    rhel7_ifaces_map = {iface.mac: iface for iface in rhel7_ifaces}
+    rhel8_ifaces_map = {iface.mac: iface for iface in rhel8_ifaces}
+
+    initrd_files = []
+    missing_ifaces = []
+    renamed_interfaces = []
+
+    if rhel7_ifaces != rhel8_ifaces:
+        for iface in rhel7_ifaces:
+            rhel7_name = rhel7_ifaces_map[iface.mac].name
+            try:
+                rhel8_name = rhel8_ifaces_map[iface.mac].name
+            except KeyError:
+                missing_ifaces.append(iface)
+                api.current_logger().warning(
+                    'The device with MAC "{}" is not detected in the upgrade'
+                    ' environment. Required driver: "{}".'
+                    ' Original interface name: "{}".'
+                    .format(iface.mac, iface.driver, iface.name)
+                )
+                continue
+
+            if rhel7_name != rhel8_name:
+                api.current_logger().warning('Detected interface rename {} -> {}.'.format(rhel7_name, rhel8_name))
+
+                if re.search('eth[0-9]+', iface.name) is not None:
+                    api.current_logger().warning('Interface named using eth prefix, refusing to generate link file')
+                    renamed_interfaces.append(RenamedInterface(**{'rhel7_name': rhel7_name,
+                                                                  'rhel8_name': rhel8_name}))
+                    continue
+
+                initrd_files.append(generate_link_file(iface))
+
+    if missing_ifaces:
+        msg = (
+            'Some network devices have not been detected inside the'
+            ' upgrade environment and so related network interfaces'
+            ' could be renamed on the upgraded system.'
+        )
+        # Note:
```

Comment:
nit: I'd honestly prefere NOTE(pstodulk) as this saves a git blame call

Reply:
To me it doesn't really matter, who made the note. It helps to understand the code.

Reply:
fixed :)